Lib/test/test_codecs.py

```diff
@@ -3415,5 +3415,28 @@ def test_rot13_func(self):
             'To be, or not to be, that is the question')
 
 
+class CodecNameNormalizationTest(unittest.TestCase):
+    """Test the normalizestring function via codecs module"""
```

Comment:
The fact that the codec names are normalized via the `normalizestring` function is an implementation detail, so the test should be written ignoring that detail, to the extent possible. This includes its documentation, so...

```suggestion
    """Test codec name normalization"""
```

Reply:
Hm, I considered this details before. Why I am still leave this detail? 
IMHO, leave enough details can help developers to maintain.

Reply:
You're welcome to add a comment after the doc-string. I wouldn't want the doc-string to mention normalizestring, though, unless we change the test to call `normalizestring()` directly instead of `codecs.lookup()`, which I don't recommend.

From my experience, however, such comments often become outdated when the implementation changes. And outdated comments causing confusion are usually worse than no comments.

Also, if someone needs to see the details, they'll likely need to go through the code starting at `codecs.lookup()` anyways. It's relatively straightforward to follow: `codecs` imports `lookup` from `_codecs`, implemented in `Modules/_codecsmodule.c`. `lookup()` is a simple wrapper for `_PyCodec_Lookup` from `Python/codecs.c`. With the imports and wrappers out of the way, `_PyCodec_Lookup` calls `normalizestring()` near the top of the function, immediately after a bit of initialization.

Reply:
> From my experience, however, such comments often become outdated when the implementation changes. And outdated comments causing confusion are usually worse than no comments.

Thanks for your share, Tal. It's make sense :)